samples/cpp/rag_sample/src/handle_master.cpp

```diff
@@ -0,0 +1,80 @@
+#include "handle_master.hpp"
+
+using json = nlohmann::json;
+
+std::function<void(const httplib::Request&, httplib::Response&)> HandleMaster::get_handle(std::string handle_name,
+                                                                                          HandleInput handle_type,
+                                                                                          util::Args args) {
+    if (auto embedding_pointer = std::get_if<std::shared_ptr<Embeddings>>(&handle_type)) {
+        if (handle_name == "embeddings_init") {
+            return get_handle_embeddings_init(*embedding_pointer, args);
+        } else {
+            return get_handle_embeddings(*embedding_pointer);
+        }
+    } else if (auto llm_pointer = std::get_if<std::shared_ptr<util::LLMPipelineUtil>>(&handle_type)) {
+        if (handle_name == "llm_init") {
+            return get_handle_llm_init(*llm_pointer, args);
+        } else {
+            return get_handle_llm(*llm_pointer);
+        }
+    }
+    return {};
+}
+
+std::function<void(const httplib::Request&, httplib::Response&)> HandleMaster::get_handle_llm_init(
+    std::shared_ptr<util::LLMPipelineUtil>& llm_pointer_ref,
+    util::Args args) {
+    const auto handle_llm_init = [&llm_pointer_ref, args](const httplib::Request& req,
+                                                          httplib::Response& res) {
+        llm_pointer_ref->pipe_pointer = std::make_shared<ov::genai::LLMPipeline>(args.llm_model_path, args.llm_device);
+        llm_pointer_ref->config = llm_pointer_ref->pipe_pointer->get_generation_config();
+        llm_pointer_ref->config.max_new_tokens = args.max_new_tokens;
+    };
+    return handle_llm_init;
+}
+
+std::function<void(const httplib::Request&, httplib::Response&)> HandleMaster::get_handle_llm(
+    std::shared_ptr<util::LLMPipelineUtil>& llm_pointer_ref) {
```

Comment:
I think we may change this to a structure like `context` to maintain following, not only limited to llm_pipeline
- pointer
- configuration
- status
But we can leave it in next PR